### PR TITLE
correctly 'copy from' and 'insert from subquery' into partitioned tables

### PR DESCRIPTION
**Pull request by mfelsche**

### CODE REVIEW

File: sql/src/main/java/io/crate/planner/projection/AbstractIndexWriterProjection.java [hunk 1]
```diff
@@ -129,44 +133,22 @@ public Integer bulkActions() {
...
-        return result;
+    @Nullable
+    public String partitionIdent() {
+        return partitionIdent;
     }
 
     @Override
     public void readFrom(StreamInput in) throws IOException {
-        tableName = in.readString();
+        TableIdent tIdent = new TableIdent();
```
Comment (mfussenegger) [inline sql/src/main/java/io/crate/planner/projection/AbstractIndexWriterProjection.java]: why the separate local variable and not use tableIdent directly?

File: sql/src/test/java/io/crate/integrationtests/InsertIntoIntegrationTest.java [hunk 2]
```diff
@@ -623,11 +624,62 @@ public void testInsertFromQueryOnDuplicateKey() throws Exception {
...
+        ensureGreen();
+        execute("insert into custom.source (name, zipcode, city) values (?, ?, ?)", new Object[][]{
+                {"Schulz", "10243", "Berlin"},
+                {"Dings", "14713", "Leipzig"},
+                {"Foo", "10243", "Musterhausen"}
+        });
+        ensureGreen();
+        refresh();
+
+        execute("select * from custom.source order by zipcode");
```
Comment (mfussenegger) [inline sql/src/test/java/io/crate/integrationtests/InsertIntoIntegrationTest.java]: is this statement here relevant for the test?

File: sql/src/test/java/io/crate/integrationtests/PartitionedTableIntegrationTest.java [hunk 3]
```diff
@@ -127,6 +136,45 @@ public void testCopyFromIntoPartitionedTable() throws Exception {
...
+        File copyFromFile = folder.newFile();
+        BufferedWriter writer = new BufferedWriter(new FileWriter(copyFromFile));
+        writer.write(
+                "{\"id\":1, \"month\":1425168000000, \"created\":1425901500000}\n" +
+                        "{\"id\":2, \"month\":1420070400000,\"created\":1425901460000}");
+        writer.flush();
+        writer.close();
+        String uriPath = Paths.get(copyFromFile.toURI()).toString();
+
+        ImmutableOpenMap<String, IndexTemplateMetaData> templates = clusterService().state().metaData().templates();
```
Comment (mfussenegger) [inline sql/src/test/java/io/crate/integrationtests/PartitionedTableIntegrationTest.java]: templates aren't used?

File: sql/src/test/java/io/crate/integrationtests/InsertIntoIntegrationTest.java (hunk 2 above)
Comment (mfelsche) [inline sql/src/test/java/io/crate/integrationtests/InsertIntoIntegrationTest.java]: jep, checking (quite verbose) that what is inside that table gets inserted (both by values and by subquery) correctly

**Comment by mfussenegger** [comment]
ack

**Pull request merged**